collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/CustomDatePickerDialog.java

```diff
@@ -242,7 +189,11 @@ public int getYear() {
     }
 
     public LocalDateTime getDate() {
-        return date;
+        if (viewModel.getDate() == null) {
```

Comment:
I think it should be viewmodel's responsibility not to return null, so this check should be moved there and here you should just call viewModel.getDate();

Reply:
Should I pass all the other arguments to the ViewModel too? Now That I have a ViewModel, it seems like I should use just one way of communicating with the fragment.

Reply:
You mean those you access via `getArguments().getSerializable` if so yeah I think it should be stored in viewmodel.

Reply:
I tried to get all those values using the ViewModel, but now I have two instances of `DateTimeViewModel` class in the dialogs. I see that you have an idea of instantiating all the widgets inside fragments, that is really cool, and it will solve this problem. Is it fine to have two instances of the ViewModel for now?

Reply:
Why do you need two viewmodels? am I missing something?

Reply:
So for passing the values using `viewmodel`, I would be needing the `viewmodel` to be created using the `FormEntryActivity` right?, that is against your idea of creating the `viewModel` using dialog context? Sorry If I'm misinterpreting things

Reply:
I think you want to pass those arguments to `CustomDatePickerDialog` using viewmodel? If so that's not what I wanted. I just said that you can store them in the existing view model (unique for every CustomDatePickerDialog) after receiving them via `getArguments().getSerializable ` so it's ok to pass them to dialog like now but then would be good to store and access via viewmodel. Is that what made confusion?

Reply:
Oh alright! I understand it now. Thanks!